### ISSUE TITLE
Citizens not spawning on reload, but when I get far from them and come back

### ISSUE THREAD

**Issue by KhrysAK47** [CitizensDev/Citizens2#2632]
The output of command `/version` on my server is: **[11:04:39 INFO]: This server is running Paper version git-Paper-783 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
You are 3 version(s) behind**
The output of command `/version citizens` on my server is: **[11:04:46 INFO]: Citizens version 2.0.28-SNAPSHOT (build 2359)
[11:04:46 INFO]: Website: http://www.citizensnpcs.co
[11:04:46 INFO]: Author: fullwall**

Hey, I want to report a issue that I recently encountered with Citizens, when I have a NPC as villager and restart the server, the NPCs behave as usual, nothing wrong about that, but when the NPC is a player and I'm near the NPCs after a reload, they don't spawn, but their holograms do.

Here is my saves.yml: https://pastebin.com/9kAHn1m8
Here is my latest console log:  https://pastebin.com/Ka6C8Cm8

The weird thing about this is that once I go away from the render distance of the chunk and come back it appears but bugged (the name appears even tho it's visibility is disabled with /npc name)

How the NPC looks after I join near it: https://imgur.com/a/vM1jX9p
How the NPC looks after I go far away and come back: https://imgur.com/a/LMqCR3s [Name shown even tho it's disabled]

I'm using Java 11.

Thanks in advance.

**Comment by mcmonkey4eva** [comment]
- Never reload. Always do a full restart of the server.
- You're not running Java 11, you're running Java 14 per that log. Java 14 is not supported. You can technically have the option to move back to Java 8, which is fully supported on MC 1.16.5, but you should probably just move to Java 16, which is fully supported and also will be required once you update to MC 1.17.x.
- You have a critical error in that log from ProtocolLib, likely caused by Java 14, likely breaking other things as a result.
- Your regular usage of `/killall` might be relevant - those types of commands can sometimes accidentally include NPCs.
- If your server is configured to be offline mode (`online-mode: false` in the `server.properties`) this breaks several features of Citizens related to processing player-type NPCs and is likely to be relevant to your issue here.
- If all the above are not sufficient to resolve the issue, try running a plugin conflict test - https://wiki.citizensnpcs.co/Plugin_Conflict aka just stop the server, temporarily remove all plugins other than Citizens, start it back up and see if it works. If it starts working after that, you know it's some plugin getting in the way, and can test re-adding them to find out which.
- If you find a specific plugin conflict was at fault, please reply to let us know which plugin caused the incompatibility. If none of the above is sufficient to resolve the issue, please reply and we'll reopen the issue to look for more possibilities.

**Comment by KhrysAK47** [comment]
Hey there,

I just did everything you said above, this includes;

1. When I say I reloaded, I meant I fully restarted the server (aka full server restart, not actual /reload)
2. Enabled online-mode in server.properties
2. Updated to Java 16
3. Updated to latest ProtocolLib

I found the conflicting plugin but the errors persist, the conflicting plugin is https://www.spigotmc.org/resources/advancedtaboverlay.78905/, I have contacted the developer and made a github issue on their page, but still no response. But this is weird, since previously by using Java 8 it worked without any issues (Citizens), after changing the Java version it started breaking.

In conclusion, what should I do? Just wait for that developer response?

**Comment by KhrysAK47** [comment]
Update: the /killall command only applied to hostile mobs, aka /killall hostile Server, meaning it did not affect the NPCs. 

I also tried using dev-builds of ProtocolLib, but didn't make much effect.

Not sure I can link the github issue here, but https://github.com/CodeCrafter47/AdvancedTabOverlay/issues/24

**Comment by mcmonkey4eva** [comment]
Oh yeah, that'd do it. It's quite likely that plugin modifies tab list packets in a way that breaks NPCs, and will need to be updated to let NPC-related packets through (Citizens has to send multiple fake tab list packets to spawn player-type NPCs, see for reference <https://wiki.citizensnpcs.co/Frequently_Asked_Questions#Why_are_NPCs_showing_up_in_the_tablist.3F>).

The short term solution if you need NPCs working *right now* would be remove that plugin from your server. (Long term solution of course being either that plugin fixes the issue, or you find an alternative that isn't bugged)

I'll comment on the issue there to offer my help with fixing compatibility.
